### PR TITLE
eCommerse Application Sprint#4

### PR DESCRIPTION
**Pull request by Elena-lucky**
1. Task: [eCommerce Application Sprint#4](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint%234.md)

2. Screenshot:
image
image
image
image

3. Deploy: [Coffee Loop](https://coffee-loop.netlify.app)

4. Done 17.06.2024 / deadline 18.06.2024

5. Score: 60 / 60

## Mentor Evaluation Criteria (60 points - Total): 👨‍🏫

✔️ Code Review Participation (10 points): Students actively participate in code reviews, providing constructive feedback and suggestions to their peers. 🗣️
✔️ No Merges Without Approvals (10 points): Pull requests are not merged without approvals from other team members, ensuring the quality of code contributions. 👍
✔️ Pull Request Descriptions (10 points): Pull requests include detailed descriptions, outlining the changes made and the purpose of the changes, facilitating better understanding and review. 📝
✔️ Task Board Management (10 points): Students maintain and regularly update a task board (e.g., Trello, Jira), ensuring that tasks are moved according to their progress and completion. 📊
✔️ Active Participation in Development Discussions (5 points): Students take an active role in discussing the development process, sharing ideas, and collaborating on solutions and improvements. 🗨️
✔️ Unit Test Coverage (15 points): The codebase has a minimum of 30% unit test coverage, ensuring the reliability and robustness of the implemented features. ✔️

## Penalties for Mentor Evaluation Criteria: ❌

❌Linter Errors (5 points): If there are errors from the linter, a penalty of 5 points will be applied. Ensure that you follow the linter rules to maintain high code quality. 🛠️📝🛑

❌ Prettier Errors (5 points): If there are formatting errors detected by Prettier, a penalty of 5 points will be applied. It's important to adhere to established formatting conventions for clean, readable code. 📐✍️🛑

❌ Presence of 'any' (5 points): The use of 'any' in TypeScript can lead to potential typing issues and is generally discouraged. A penalty of 5 points will be applied if 'any' is found in the code. 🚫🔍🛑

❌ Non-functional Scripts in package.json (5 points): All scripts defined in the package.json should function correctly. If there are non-working scripts, a penalty of 5 points will be applied. ⚙️📦🛑

❌ Non-SPA Application (60 points): The application should be a Single Page Application (SPA) with an empty index.html. If this requirement is not met, a penalty of 60 points will be applied. ⚠️🖥️🛑

### CODE REVIEW

**Review by egorokunevich — APPROVED** [review]
Nice job! Thanks!

File: src/pages/AboutUsPage/AboutUsPage.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    const descriptionElement = p({ className: styles.devDescription, txt: dev.description });
+    const funFactElement = p({ className: styles.devFunFact, txt: `Note! ${dev.funFact}` });
+
+    infoContainer.append(nameLink, roleElement, educationElement, descriptionElement, funFactElement);
+    devCard.append(photoContainer, infoContainer);
+
+    return devCard;
+  }
+
+  private getDevelopers(): Developer[] {
```
Comment (madkorney) [inline src/pages/AboutUsPage/AboutUsPage.ts:81]: great implementation!   however it makes sense to move the data to separate file - as a constant or as a json and import it here. to split the content and rendering into separate entities

File: src/pages/CatalogPage/CatalogPage.ts [hunk 2]
```diff
@@ -379,50 +390,23 @@ export class CatalogPage {
...
-      this.updateCards(products);
+      if (products) {
+        this.updateCards(products);
+      }
     });
     container.append(title, selectContainer);
     return container;
   }
 
-  // private createTimeAttributeFilter(): HTMLElement {
```
Comment (madkorney) [inline src/pages/CatalogPage/CatalogPage.ts:388]: commented code

**Review by madkorney — COMMENTED** [review]
Good job! solid code style, great naming. very well done.

**Pull request merged**